testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -147,3 +151,29 @@ def test_density_from_Universe_update_selection(self):
             self.selections['dynamic'],
             self.references['dynamic']['meandensity'],
             update_selections=True)
+
+class TestGridImport(TestCase):
+
+    def setUp(self):
+        sys.modules.pop('MDAnalysis.analysis.density', None)
+
+    @block_import('gridData')
+    def test_absence_griddata(self):
+        # if gridData package is missing an ImportError should be raised
+        # at the module level of MDAnalysis.analysis.density
+        with assert_raises(ImportError):
+            import MDAnalysis.analysis.density
+
+    def test_presence_griddata(self):
+        # no ImportError exception is raised when gridData is properly
+        # imported by MDAnalysis.analysis.density
+
+        # mock gridData in case there are testing scenarios where
+        # it is not available
```

Comment:
Here is what the CHANGELOG entry says in version 0.7.5:

> * install a range of analysis dependencies right away: networkx,
    biopython, GridDataFormats (usually all painless); leave scipy and
    matplotlib to the user and the local package manager

The reasoning makes sense to me: having the easy dependencies already installed makes a better user experience. If a user wants to calculate a density, he does not have to install anything new. scipy and matplotlib are trickier beasts to install, they are also much heavier; having them optional seems reasonable, the same goes for scikit-learn that depends on scipy.

I know there packages are easier to install with conda. But we should not assume everybody uses conda.